src/Adapter/Product/Repository/CombinationRepository.php

```diff
@@ -0,0 +1,253 @@
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+    }
+
+    /**
+     * @param Combination $combination
+     *
+     * @return CombinationId
+     *
+     * @throws CombinationConstraintException
+     * @throws CoreException
+     */
+    public function add(Combination $combination): CombinationId
+    {
+        //@todo: validation?
+        $id = $this->addObjectModel($combination, CannotAddCombinationException::class);
+
+        return new CombinationId($id);
+    }
+
+    /**
+     * @param int $productId
+     * @param int $limit
+     * @param int $offset
+     * @param array $filters
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    public function getProductCombinations(int $productId, ?int $limit = null, ?int $offset = null, array $filters = []): array
+    {
+        $qb = $this->getCombinationsQueryBuilder($productId, $filters)
+            ->select('pa.*')
+            ->setParameter('productId', $productId)
+        ;
+
+        if ($offset) {
+            $qb->setFirstResult($offset);
+        }
+
+        if ($limit) {
+            $qb->setMaxResults($limit);
+        }
+
+        return  $qb->execute()->fetchAll();
+    }
+
+    /**
+     * @param int $productId
+     * @param array $filters
+     *
+     * @return int
+     */
+    public function getTotalCombinationsCount(int $productId, array $filters = []): int
+    {
+        $qb = $this->getCombinationsQueryBuilder($productId, $filters)->select('COUNT(pa.id_product_attribute) AS total_combinations');
+
+        return (int) $qb->execute()->fetch()['total_combinations'];
+    }
+
+    /**
+     * @param int[] $combinationIds
+     * @param LanguageId $langId
+     *
+     * @return array<int, array<int, mixed>>
+     */
+    public function getAttributesInfoByCombinationIds(array $combinationIds, LanguageId $langId): array
+    {
+        $attributeCombinationAssociations = $this->getAttributeCombinationAssociations($combinationIds);
+
+        $attributeIds = array_unique(array_map(function ($attributeByCombination) {
+            return $attributeByCombination['id_attribute'];
+        }, $attributeCombinationAssociations));
+
+        $attributesInfoByAttributeId = $this->getAttributesInformation($attributeIds, $langId->getValue());
+
+        $attributesInfoByCombinationId = [];
+        foreach ($attributeCombinationAssociations as $attributeCombinationAssociation) {
+            $combinationId = (int) $attributeCombinationAssociation['id_product_attribute'];
+            $attributeId = (int) $attributeCombinationAssociation['id_attribute'];
+            $attributesInfoByCombinationId[$combinationId] = $attributesInfoByAttributeId[$attributeId];
```

Comment:
```suggestion
            $attributesInfoByCombinationId[$combinationId][] = $attributesInfoByAttributeId[$attributeId];
```
Shouldn't it be a list, since a combination can have multiple attributes?